Jbtm 2814 tx vertx

mmusgrov opened the pull request:
https://issues.jboss.org/browse/JBTM-2814
!BLACKTIE !XTS !QA_JTA !QA_JTS_JACORB !QA_JTS_JDKORB !PERF

mmusgrov commented:
@tomjenkinson @nmcl has already reviewed this PR but I want to give you the opportunity to comment before I merge.

Note that in a future PR it may make sense to put some of the code in the quickstart repo but for now I would like it to be in the main repo so that we have an easily accessible audit trail.

Also I think the inclusion of transactional vertx in the main repo is worth a blog post before you put out the next release so I will schedule that task.

tomjenkinson commented on vertx/echo/EchoClient.java:
```diff
@@ -1,25 +1,7 @@
-/*
```
Why copyright removed?

mmusgrov replied:
The original was taken from the vertx repo and the examples do not contain copyrights.

tomjenkinson commented on vertx/echo/EchoServer.java:
```diff
@@ -1,34 +1,30 @@
+import io.vertx.core.streams.Pump;
+
 /*
- * Copyright 2011 the original author or authors.
```
Why copyright removed?

mmusgrov replied:
ditto

tomjenkinson commented on vertx/echo/build.sh:
```diff
@@ -0,0 +1,29 @@
+#JAVA_HOME=/usr/local/jdk1.8.0_91
```
add copyright

mmusgrov replied:
ditto

tomjenkinson commented on vertx/echo/pom.xml:
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
add copyright

mmusgrov replied:
Although I created this pom, to be consistent with the original vert.x example, there should be no copyright.

tomjenkinson commented on vertx/module/pom.xml:
```diff
@@ -59,134 +63,84 @@
+    <dependency>
+      <groupId>org.jboss.logging</groupId>
+      <artifactId>jboss-logging</artifactId>
+      <version>3.1.4.GA</version>
```
declare version in properties

tomjenkinson commented on vertx/raw/pom.xml:
```diff
@@ -0,0 +1,111 @@
+    <dependency>
+      <groupId>org.jboss.logging</groupId>
+      <artifactId>jboss-logging</artifactId>
+      <version>3.1.4.GA</version>
```
version from properties

tomjenkinson commented:
There are quite a few files with copyright notice removed - is that intentional? There are files you added that don't have a copyright notice.

tomjenkinson commented:
That was @mmusgrov

mmusgrov commented:
@tomjenkinson:

1. I did not remove any copyright notices - but I did need to take a copy of the echo example from the Vert.x 3 repo and those examples no longer contain copyrights
2. I added a version property for referencing ${jboss-logging.version}
3. I added copyrights to the poms that I added

tomjenkinson commented:
Makes sense - thanks

tomjenkinson commented:
RfM imo

tomjenkinson commented:
REST-AT failure caused by WFLY-7785

mmusgrov commented:
Yes I noticed that you had drilled down into the root case with Gytis. 

I will merge then
